network/phonebook/phonebook.go

```diff
@@ -29,23 +29,84 @@ import (
 // of how many addresses the phonebook actually has. ( with the retry-after logic applied )
 const getAllAddresses = math.MaxInt32
 
-// PhoneBookEntryRoles defines the roles that a single entry on the phonebook can take.
+// RoleSet defines the roles that a single entry on the phonebook can take.
 // currently, we have two roles : relay role and archival role, which are mutually exclusive.
 //
-//msgp:ignore PhoneBookEntryRoles
-type PhoneBookEntryRoles int
+//msgp:ignore Roles
+type RoleSet struct {
+	roles       Role
+	persistence persistence // parallel bits indicating which roles are persistent
```

Comment:
You might use the `Role` type to track persistence too, to decrease the amount of stuff to understand.  If the bit is set in `roles`, you have the role. If it's set in `persistence` it would be persistent. But they are both using a `Role` to hold bits.